File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String temp = "";
        
        while(n != 0){
            temp = n%k + temp;
            n/=k;
        } // k진법 변환
        String[] arr = temp.split("0");
        for(String a : arr){
            if(a.equals("")){
                continue;
            }
            long num = Long.parseLong(a);
            
            if(isPrime(num)){
                answer++;
            }
        }
        
        
        
        
        return answer;
    }
    
    public static boolean isPrime(long num) {
        if(num<2){
            return false;
        }
        for(int i = 2; i <= Math.sqrt(num); i++){
            if(num%i == 0){
                return false;
            }
        }
        return true;
    }
    
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def solution(n, k):
    answer = 0
    num = ''
    while n > 0:
        remainder = n % k
        n = n // k
        num = str(remainder)+num
    
    num_list = num.split("0")
    num_list = [x for x in num_list if x!=""]
    
    for i in range(len(num_list)):
        cnt = 0
        if num_list[i] == "1":
            cnt += 1
        else:
            for j in range(2, int(int(num_list[i])**0.5)+1):
                if int(num_list[i])%j==0:
                           cnt+=1
                           break
        if cnt == 0:
            answer+=1
    
    return answer